sopel/bot.py

```diff
@@ -394,16 +398,17 @@ def get_plugin_meta(self, name):
         version='7.1',
         removed_in='8.0')
     def unregister(self, obj):
```

Comment:
In hindsight, I'm not sure removing functionality from this method bit by bit was the right approach. We're left with a "deprecated" method that actually won't do what it used to any more, so it's no longer backward compatible.

This is partly on me for not thinking of it before, but since this is kind of a bot internal… should we just delete this now and skip the deprecation cycle? It's either that or restore all functionality to work just like it did in 7.0.4 (and earlier), and maintain it up until 8.0.

Reply:
Hm yeah we should remove it entirely. I didn't do it at first because... I don't know? Maybe I removed them bit by bit in so many different PRs that I didn't think about it too much.